### ISSUE TITLE
0.7.10 breaks/removes type checking for parameters of all request methods

### ISSUE THREAD

**Issue by iToXiQ** [drwpow/openapi-typescript#1384]
After updating to 0.7.10 required path and body params are always seen as optional in typescript.

i.E.
```
import createClient from 'openapi-fetch';
// sdk with a get /user/{id} path, which requires path.params.id
import { paths } from "my-sdk";

const client = createClient<paths>();

// typescript should be complaining because no path.params.id was provided
await client.get("/user/{id}");

// same thing applies here
await client.get("/user/{id}", {});

// this happens for all http request methods
```

Versions:
- Typescript 5.1.6
- Node 18.9.0
- openapi-fetch 0.7.10

**Comment by drwpow** [comment]
Ah this is good to know! Will add a note in the docs recommending this setup for TypeScript.

**Comment by iToXiQ** [comment]
Would it be possible to make this this library compatible for ts projects with "strictNullChecks": false?
i.E. my closed PR where i swapped "undefined extends T" for sth that works with both ts config setups?

**Comment by drwpow** [comment]
Yeah will take a 2nd look at that! For sure I’m onboard with any improvements that don’t make adoption dependent on a specific TSConfig. Granted, I don’t think supporting _every possible TSConfig_ is realistic. But we should shoot for supporting as many different TSConfig settings as possible.
